### PR TITLE
docs: Autodoc typehints

### PR DESCRIPTION
**Pull request by BobDotCom**
## Summary

Automatically use typehints to generate docs.

## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Comment by BobDotCom** [comment]
Note: ATM, we cannot set return descriptions without setting a return type

**Comment by BobDotCom** [comment]
Most/all return type annotations should be removed in the future, once it's supported

**Review by Lulalaby — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Comment by BobDotCom** [comment]
> Most/all return type annotations should be removed in the future, once it's supported

See sphinx-doc/sphinx#10134

File: discord/client.py [hunk 1]
```diff
@@ -721,13 +721,16 @@ def stop_loop_on_completion(f):
...
 
     @property
     def activity(self) -> ActivityTypes | None:
-        """Optional[:class:`.BaseActivity`]: The activity being used upon
-        logging in.
+        """The activity being used upon logging in.
+
+        Returns
+        -------
+        Optional[:class:`.BaseActivity`]
```
Comment (EmreTech) [inline discord/client.py:733]: Why is this out of line with the actual return type annotation of this property? Is there a good reason, and if so, could it be fixed in another PR?

Comment (BobDotCom) [reply discord/client.py:733]: It's just keeping the original type, maybe could be changed in the future

**Comment by Dorukyum** [comment]
Has this been tested?

**Review by Dorukyum — APPROVED** [review]
Asked if this was tested because you hadn't checked that in the checklist

**Review by Middledot — APPROVED** [review]
